nettacker/core/lib/ssl.py

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env python
+
+import logging
+import socket
+import ssl
+from datetime import datetime, timezone
+
+from OpenSSL import crypto
+
+from nettacker.core.lib.base import BaseEngine, BaseLibrary
+
+log = logging.getLogger(__name__)
+
+
+def is_weak_hash_algo(algo):
+    algo = algo.lower()
+    for unsafe_algo in ("md2", "md4", "md5", "sha1"):
+        if unsafe_algo in algo:
+            return True
+    return False
+
+
+def is_weak_ssl_version(host, port, timeout):
+    def test_ssl_verison(host, port, timeout, ssl_version=None):
+        try:
+            context = ssl.SSLContext(ssl_version)
+            socket_connection = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            socket_connection.settimeout(timeout)
+            socket_connection.connect((host, port))
```

Comment:
Could you extract it to a separate function and reuse it instead (reduce the code repetition)?